lib/spec/collection.js

```diff
@@ -34,13 +34,13 @@ export class Collection extends Spec {
       }
     }
 
-    const ret = [];
-    for (const [key, val] of value.entries()) {
-      const conformed = dt(this.options.spec, val);
-      if (conformed === invalid) {
-        return invalid;
-      }
-      ret[key] = conformed;
+    const spreadValue = p.array(value) ? value : [...value];
```

Comment:
for / of was adding in a `length` key, which was the source of this bug. Using spread operator converts the Set to an array with simpler map / filter functions.